src/Domain/UseCase/VerifyMFA.ts

```diff
@@ -26,32 +27,33 @@ export class VerifyMFA implements UseCaseInterface {
       }
     }
 
-    const mfaExtension = await this.itemsRepository.findMFAExtensionByUserUuid(user.uuid)
+    const mfaSecretSetting = await this.settingRepository.findOneByNameAndUserUuid(SETTINGS.MFA_SECRET, user.uuid)
 
-    if (!mfaExtension) {
+    if (!mfaSecretSetting) {
       return {
         success: true
       }
     }
 
-    const mfaParamKey = `mfa_${mfaExtension.uuid}`
-    if (!dto.requestParams[mfaParamKey]) {
+    if (!dto.token) {
       return {
         success: false,
         errorTag: 'mfa-required',
-        errorMessage: 'Please enter your two-factor authentication code.',
-        errorPayload: { mfa_key: mfaParamKey }
+        errorMessage: 'Please enter your two-factor authentication code.'
       }
     }
 
-    const mfaContent = this.contentDecoder.decode(mfaExtension.content)
+    const decryptedValue = await this.crypter.decrypt(
+      mfaSecretSetting.encryptionVersion,
+      mfaSecretSetting.value,
+      <string> user.serverKey
```

Comment:
This is the user's `encryptedServerKey`. At what point is the `encryptedServerKey` decrypted by the global encryption key? I can't seem to find where that's happening, as it doesn't seem to be happening inside of Cryper.ts.